Makefile

```diff
@@ -116,6 +116,24 @@ NEWSBOAT_LIBS+=-liconv -lintl
 PODBOAT_LIBS+=-liconv -lintl
 endif
 
+# All text files except for temporary editor files and binaries will be checked
+# for trailing whitespace.
+CHECK_TRAILING_WHITESPACE_FILES=$(shell find . -type f ! \( \
+	   -path '*/.git/*' -prune \
+	-o -path '*/target/*' -prune \
+	-o -path '*/doc/generate' \
+	-o -path '*/doc/generate2' \
+	-o -path '*/doc/gen-example-config' \
+	-o -path '*/test/test' \
+	-o -name 'newsboat' \
+	-o -name 'podboat' \
+	-o -name '*.mo' \
+	-o -name '*.o' \
+	-o -name '*.a' \
+	-o -name '*.sw?' \
+	-o -name '*~' \
+	\) -print)
+
```

Comment:
This list looks similar to .gitignore. Instead of maintaining two lists, how about replacing this with `git ls-files`?

Reply:
I thought so as well, but then it occurred to me if somebody obtains a tarball and has no git, formatting would fail. What do you reckon, just bad luck for those people?

Reply:
Oh, good point, I didn't think of that! I don't think the tarball is a good base for development: it doesn't include some files (see .gitattributes), and it lags behind `master` which can lead to conflicts when a patch is submitted. Given how rarely I get patches over email, I think it's okay to just say "please make a Git clone if you want to do development". So please replace this with `git ls-files`.

Reply:
Fair enough. :-) I changed it to `git ls-files` and excluded all the test data because there seems to be some non-text files, like mp3 and things like that. But I can also just exclude those specific file extensions that seem to hold binary data.

Reply:
Aaaaand I just realized, that all those are actually empty files and not real binary ones. Let me remove this filter.

Reply:
Now that should do the trick!

I ran out of screen time today, will verify the expected CI failure tomorrow and then also remove the test commit for it.

Reply:
Looking good, removing the test commit. CI should now pass.